xwiki-platform-core/xwiki-platform-oldcore/src/test/java/com/xpn/xwiki/web/DownloadActionTest.java

```diff
@@ -768,4 +774,137 @@ void downloadWhenEmptyRange() throws XWikiException, IOException
         verifyOutputExpectations(0, this.fileContent.length);
         verifyResponseExpectations(d.getTime(), this.fileContent.length);
     }
+
+    private static class StubServletOutputStream extends ServletOutputStream
```

Comment:
Out of curiosity, why a stub and not a mock?

Reply:
So here I just copy/pasted the tests that were in MockitoDownloadActionTest, I fixed some obvious stuff, but I didn't spent much time thinking about that one.